hls-test-utils/src/Test/Hls/Util.hs

```diff
@@ -322,38 +314,31 @@ expectNoMoreDiagnostics timeout doc src = do
 -- If timeout is 0 it will wait until the session timeout
 waitForDiagnosticsFromSourceWithTimeout :: Seconds -> TextDocumentIdentifier -> String -> Test.Session [Diagnostic]
 waitForDiagnosticsFromSourceWithTimeout timeout document source = do
-    when (timeout > 0) $ do
+    when (timeout > 0) $
         -- Give any further diagnostic messages time to arrive.
         liftIO $ sleep timeout
         -- Send a dummy message to provoke a response from the server.
         -- This guarantees that we have at least one message to
         -- process, so message won't block or timeout.
-        void $ Test.sendNotification (SCustomMethod "non-existent-method") A.Null
-    handleMessages
+    testId <- Test.sendRequest (SCustomMethod "test") (A.toJSON GetFilesOfInterest)
+    handleMessages testId
```

Comment:
this sounds like it is going to perform some actual work (although very minor), seems dubious to me?

Reply:
I adjusted the indentation to 4 spaces so the diff is messier. This is previously `SCustomMethod "non-existent-method"`, and the client waits for a logging diagnostics from the server, which is no longer present. I think instead of adding the log back, change the test to not rely on a specific form of logger is better.

And there is a similar function in `ghcide` which is newer and doesn't rely on logs to function properly:
https://github.com/haskell/haskell-language-server/blob/347a7187f20242540926b9927c59a45b18798c67/ghcide/test/src/Development/IDE/Test.hs#L92-L101